src/ui/public/inspector/adapters/data_adapter.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EventEmitter } from 'events';
+
+// TODO: add a more specific TabularData type.
+type TabularData = any;
```

Comment:
It would make sense, to have a proper typing for that tabular data. But since we are starting to use that tabular data with the pipeline now in more places and we're having already tabify I would rather take some time to come up with a proper solution of that tabular data formats and how we want to interchange them. Since we are not losing anything by saying `any` here over using JavaScript in general, I would postpone that overall typescript tabular thoughts to another PR.